apitools/base/protorpclite/messages.py

```diff
@@ -1060,6 +1058,10 @@ def __ne__(self, other):
         """
         return not self.__eq__(other)
 
+    def __hash__(self):
+        """Hash by __tags"""
+        return hash(self.__tags)
```

Comment:
Looking closer at this, `self.__tags` is a dictionary, which is unhashable. Looks like we have two options:
- Keep the class unhashable. 
- Hash over contents of `__tags` or it's subset, if we can consider Message objects immutable, which we probably cannot.

Is there a reason to believe Message objects are immutable? If not, I think making them unhashable makes most sense.

Reply:
`self.__tags` is a dict! I missed that, sorry :man_facepalming: 

I think Message should be mutable. Therefore, it should be unhashable.
This class is hashable in Python 2. Making it unhashable would break compatibility in Python 3. But, that is also an option. 

If we make it unhashable, we should add `__hash__ = None` to the class to keep the behaviour consistent in Python 2 as well.
OR
If we want to keep behaviour consistent in Python 3, while not breaking compatibility, we should add `__hash__ = object.__hash__` to the class.

Reply:
There are good reasons why in Python 3 `__hash__` is not implemented by default, in this light, I think the better way to keep compatibility would be to set `__hash__ = None`. This may require cascading changes for Py2 users of `apitools`. But same/similar changes would still be needed in downstream codebase to make relevant code Python3-compatible, so that may encourage users of apitools to fix their code sooner than later.

Reply:
Cool. I'll make that change :)